### ISSUE TITLE
Stuck at login

### ISSUE THREAD

**Issue by jdshkolnik** [hackjutsu/Lepton#346]
The more we know about your system and use case, the more easily and likely we can help.

#### Environment info
* OS: Windows 10 Enterprise 1803
* App Version (`Command/Ctrl + ,`): 1.7.3

#### Description of the problem / feature request / question:
I'm unable to log in on my work laptop; it just sits at the login progress bar. I've tried reinstalling with no luck. I was able to log in on my personal machine without a problem.

#### If possible, provide a sample gist or screenshot:
![Lepton login](https://user-images.githubusercontent.com/2047520/54219731-6ded6a80-44c6-11e9-9fae-37a049baadba.png)

#### If possible, provide the log files

> 2019-03-12T16:50:22.725Z - info: 
> 
> ----- Lepton v1.7.3 win32-----
> 
> 2019-03-12T16:50:22.727Z - info: [conf] Looking for .leptonrc at C:\Users\jshkolnik/.leptonrc
> 2019-03-12T16:50:22.727Z - info: [conf] The resolved configuration is ...
> 2019-03-12T16:50:22.727Z - info: "userPanel": {"hideProfilePhoto":false}
> 2019-03-12T16:50:22.727Z - info: "logger": {"level":"debug"}
> 2019-03-12T16:50:22.727Z - info: "proxy": {"enable":false,"address":"socks://localhost:1080"}
> 2019-03-12T16:50:22.728Z - info: "snippet": {"sorting":"updated_at","sortingReverse":true,"expanded":true,"newSnippetPrivate":false}
> 2019-03-12T16:50:22.728Z - info: "editor": {"tabSize":4}
> 2019-03-12T16:50:22.728Z - info: "enterprise": {"enable":false,"host":"","token":"","avatarUrl":""}
> 2019-03-12T16:50:22.728Z - info: "type": "literal"
> 2019-03-12T16:50:22.861Z - info: Checking for update
> 2019-03-12T16:50:23.089Z - debug: -----> registering login-page-ready listener
> 2019-03-12T16:50:26.137Z - info: Update for version 1.7.3 is not available (latest version: 1.7.3, downgrade is disallowed).
> 2019-03-12T16:50:26.138Z - debug: checkForUpdatesAndNotify called, downloadPromise is null
> 2019-03-12T16:50:27.308Z - debug: -----> Inside getCachedUserInfo
> 2019-03-12T16:50:27.314Z - debug: -----> [false] cachedProfile is undefined
> 2019-03-12T16:50:27.319Z - debug: -----> [false] cachedToken is undefined
> 2019-03-12T16:50:27.342Z - debug: -----> Inside LoginPage componentWillMount with loggedInUserInfonull
> 2019-03-12T16:50:27.344Z - debug: -----> Registering listener for auto-login signal
> 2019-03-12T16:50:27.345Z - debug: -----> sending login-page-ready signal
> 2019-03-12T16:50:27.346Z - info: [signal] sending auto-login signal
> 2019-03-12T16:50:27.493Z - debug: -----> Received "auto-login" signal with loggedInUserInfo null
> 2019-03-12T16:50:29.156Z - debug: -----> Inside launchAuthWindow with token undefined
> 2019-03-12T16:50:29.215Z - debug: loading authUrl https://github.com/login/oauth/authorize?client_id=xxx&scope=gist
> 2019-03-12T16:50:29.287Z - info: [Dispatch] updateAuthWindowStatus ON
> 2019-03-12T16:50:40.910Z - info: [Dispatch] updateAuthWindowStatus OFF
> 2019-03-12T16:50:40.917Z - info: [Dispatch] updateUserSession IN_PROGRESS
> 2019-03-12T16:50:40.927Z - debug: [REST] Exchanging authCode with access token
> 2019-03-12T16:50:41.078Z - error: Failed: {"code":"SELF_SIGNED_CERT_IN_CHAIN"}

**Comment by jdshkolnik** [comment]
I just noticed this is identical to #250

**Comment by jdshkolnik** [comment]
It's something to do with strict SSL (as mentioned [here](https://stapp.space/fight-with-2/)) but I don't know how to apply it to the login.

**Comment by hackjutsu** [comment]
Since it's duplicate. I'll reopen https://github.com/hackjutsu/Lepton/issues/250 and close this one. Thanks for reporting!

**Comment by estysdesu** [comment]
I have the same issue I think on my work pc. 
```
2019-06-11T01:21:39.399Z - info: 

----- Lepton v1.8.0 win32-----

2019-06-11T01:21:39.400Z - info: [conf] Looking for .leptonrc at C:\Users\<username>/.leptonrc
2019-06-11T01:21:39.400Z - info: [conf] The resolved configuration is ...
2019-06-11T01:21:39.400Z - info: "theme": "dark"
2019-06-11T01:21:39.400Z - info: "userPanel": {"hideProfilePhoto":false}
2019-06-11T01:21:39.400Z - info: "logger": {"level":"debug"}
2019-06-11T01:21:39.400Z - info: "proxy": {"enable":true,"address":"https://<hostname:port>"}
2019-06-11T01:21:39.400Z - info: "snippet": {"sorting":"updated_at","sortingReverse":true,"expanded":true,"newSnippetPrivate":false}
2019-06-11T01:21:39.400Z - info: "editor": {"tabSize":4}
2019-06-11T01:21:39.400Z - info: "enterprise": {"enable":false,"host":"","token":"","avatarUrl":""}
2019-06-11T01:21:39.400Z - info: "type": "literal"
2019-06-11T01:21:39.438Z - info: Checking for update
2019-06-11T01:21:39.607Z - debug: -----> registering login-page-ready listener
2019-06-11T01:21:40.999Z - info: [.leptonrc] Use proxy https://<hostname:port>
2019-06-11T01:21:42.509Z - debug: -----> Inside getCachedUserInfo
2019-06-11T01:21:42.513Z - debug: -----> [false] cachedProfile is undefined
2019-06-11T01:21:42.517Z - debug: -----> [false] cachedToken is undefined
2019-06-11T01:21:42.534Z - debug: -----> Inside LoginPage componentWillMount with loggedInUserInfonull
2019-06-11T01:21:42.535Z - debug: -----> Registering listener for auto-login signal
2019-06-11T01:21:42.535Z - debug: -----> sending login-page-ready signal
2019-06-11T01:21:42.536Z - info: [signal] sending auto-login signal
2019-06-11T01:21:42.619Z - debug: -----> Received "auto-login" signal with loggedInUserInfo null
2019-06-11T01:21:42.800Z - info: Update for version 1.8.0 is not available (latest version: 1.8.0, downgrade is disallowed).
2019-06-11T01:21:42.801Z - debug: checkForUpdatesAndNotify called, downloadPromise is null
2019-06-11T01:21:43.890Z - debug: -----> Inside launchAuthWindow with token undefined
2019-06-11T01:21:43.912Z - debug: loading authUrl https://github.com/login/oauth/authorize?client_id=9e42611bb02db4547abf&scope=gist
2019-06-11T01:21:43.936Z - info: [Dispatch] updateAuthWindowStatus ON
2019-06-11T01:21:51.223Z - info: [Dispatch] updateAuthWindowStatus OFF
2019-06-11T01:21:51.232Z - info: [Dispatch] updateUserSession IN_PROGRESS
2019-06-11T01:21:51.245Z - debug: [REST] Exchanging authCode with access token
2019-06-11T01:21:51.357Z - error: Failed: {"code":"ECONNRESET"}
2019-06-11T01:23:03.477Z - info: The app window is closed
```
I hid the output of the proxy and user in the logs for privacy.
